changelogs/head.asciidoc

```diff
@@ -21,3 +21,4 @@ https://github.com/elastic/apm-server/compare/8.12\...main[View commits]
 - Add support for Otel code.stacktrace {pull}12096[12096]
 - `http.request.body.bytes` now reports the bytes read from request body even if Content-Length is -1 {pull}12451[12451]
 - Use `messaging.system` and `messaging.operation` to detect OTel messaging span for transactions {pull}12506[12506]
+- Add support for OTel data stream routing by parsing `data_stream.dataset` and `data_stream.namespace` from resource attributes, instrumentation scope attributes, and signal-level attributes (in increasing order of precedence). This is a breaking change only for users relying on the existing `data_stream` attributes being stored as labels. {pull}12578[12578]
```

Comment:
I updated the PR to move the entry from Breaking Changes to Added since this isn't really super breaking. @axw are you happy with that? Any comments on the wording?

Reply:
Is "in increasing order of precedence" ambiguous?

Reply:
That sounds reasonable to me.